streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamTask.java

```diff
@@ -229,22 +240,24 @@ private void updateProcessorContext(final StampedRecord record, final ProcessorN
 
     /**
      * <pre>
-     *  - flush state and producer
-     *  - write checkpoint
-     *  - commit offsets
+     * - flush state and producer
+     * - if(!eos) write checkpoint
+     * - commit offsets and maybe start new transaction
      * </pre>
      */
     @Override
-    public void commit() {
+    public void commit(final boolean startNewTransaction) {
```

Comment:
This parameter looks not general enough to me: it is only related to eos and hence only useful when it is turned on, and the same for its inner `commitOffsets` call. I'd suggest make the public api still param-free and have a private function with the param, and having the public function call it with param set to eosEnabled, and its inner `suspend` can the just call the private one with param set to false.